File: bulls-and-cows.cpp
int vis[10];
class Solution {
public:
    string getHint(string secret, string guess) {
        for (int i = 0; i < 10; ++i) {
            vis[i] = 0;
        }
        int cows = 0, bulls = 0;
        for (int i = 0; i < secret.size(); ++i) {
            if (secret[i] == guess[i]) {
                ++bulls;
            } else {
                ++vis[secret[i] - '0'];
            }
        }
        for (int i = 0; i < guess.size(); ++i) {
            if (secret[i] != guess[i]) {
                if (vis[guess[i] - '0'] > 0) {
                    ++cows;
                    --vis[guess[i] - '0'];
                }
            }
            
        }
        char tmp[100];  
        sprintf(tmp, "%dA%dB", bulls, cows);
        return tmp;
    }
};

File: src/main/java/BullsAndCows.java
public class BullsAndCows {
    public String getHint(String secret, String guess) {
        int bulls = 0, cows = 0;
        int[] vis = new int[10];
        for (int i = 0; i < secret.length(); ++i) {
            vis[i] = 0;
        }
        for (int i = 0; i < secret.length(); ++i) {
            if (secret.charAt(i) == guess.charAt(i)) {
                ++bulls;
            } else {
                ++vis[secret.charAt(i) - '0'];
            }
        }
        for (int i = 0; i < guess.length(); ++i) {
            if (secret.charAt(i) != guess.charAt(i)) {
                if (vis[guess.charAt(i) - '0'] > 0) {
                    --vis[guess.charAt(i) - '0'];
                    ++cows;
                }
            }
        }
        String res = "" + bulls + "A" + cows + "B";
        return res;
        
    }
}
